modules/sfp_massdns.py

```diff
@@ -0,0 +1,405 @@
+        "concurrent_resolvers": "Maximum concurrent lookup threads. Bandwidth cost is ~1Mbps per 100 resolvers."
+    }
+
+    _ipRegex = re.compile(r"[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}")
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.sf.debug("Setting up sfp_massdns")
+        self.state = self.tempStorage()
+        self.state.update({
+            "sub_wordlist": [],
+            "sent_events": [],
+            "handled_events": []
+        })
+        self.__dataSource__ = "DNS"
+        self.iteration = 0
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+        self.word_regex = re.compile(r'[^\d\W_]+')
+        self.word_num_regex = re.compile(r'[^\W_]+')
+        self.num_regex = re.compile(r'\d+')
+
+        dicts_dir = f"{self.sf.myPath()}/spiderfoot/dicts"
+        if self.opts["large_wordlist"]:
+            subdomain_dict = f"{dicts_dir}/massdns-subdomains-1.2M.txt"
+        else:
+            subdomain_dict = f"{dicts_dir}/massdns-subdomains-110K.txt"
+        with open(subdomain_dict, "r") as f:
+            self.state["sub_wordlist"] = list(set([x.strip().lower() for x in f.readlines()]))
+        with open(f"{dicts_dir}/subdomain-mutations.txt", "r") as f:
+            if self.opts["alphamutation"]:
+                self.state["alpha_mutation_wordlist"] = list(set([x.strip().lower() for x in f.readlines()]))
+
+        # set up nameservers
+        self.resolvers = []
+        min_reliability = .99
+        nameservers = set()
+        nameservers_url = "https://public-dns.info/nameserver/nameservers.json"
+        nameservers_dict = self.sf.myPath() + "/dicts/resolvers.txt"
+        # get every valid nameserver
+        try:
+            fetched_nameservers = json.loads(self.sf.fetchUrl(
+                nameservers_url,
+                useragent=self.opts.get("_useragent", "Spiderfoot")
+            )["content"])
+        except ValueError:
+            fetched_nameservers = []
+        for entry in fetched_nameservers:
+            ip_address = str(entry.get("ip", "")).strip()
+            try:
+                reliability = float(entry.get("reliability", 0))
+            except ValueError:
+                continue
+            if reliability >= min_reliability and self._ipRegex.match(ip_address):
+                nameservers.add(ip_address)
+        # fall back to local dict if necessary
+        if not nameservers:
+            self.sf.debug(f"Failed to retrieve nameservers from {nameservers_url}")
+            nameservers = set(self._ipRegex.findall(open(nameservers_dict, "r").read()))
+            self.sf.debug(f"Loaded {len(nameservers):,} nameservers from {nameservers_dict}")
+        else:
+            self.sf.debug(f"Loaded {len(nameservers):,} nameservers from {nameservers_url}")
+        self.resolvers = self.verifyNameservers(nameservers)
+        self.sf.info(f"Using {len(self.resolvers):,} valid nameservers")
```

Comment:
The contents of the `resolvers.txt` file is likely to change frequently.

Rather than ship `resolvers.txt` with SpiderFoot, this functionality should be replaced with cacheGet and cachePut.